integrationtests/tests/tests.rs

```diff
@@ -970,3 +971,53 @@ async fn rejoin_consensus_threshold_peers() -> Result<()> {
     })
     .await
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn ecash_can_be_recovered() -> Result<()> {
```

Comment:
Wasn't this code in a separate PR?

Reply:
Yes, I've included it as a commit in this PR, because otherwise there's nothing actually testing if my changes here work. :D